api/src/wfl/sink.clj

```diff
@@ -171,12 +196,198 @@
       (util/select-non-nil-keys (keys terra-workspace-sink-serialized-fields))
       (assoc :name terra-workspace-sink-name)))
 
-(defoverload stage/validate-or-throw terra-workspace-sink-name verify-terra-sink!)
-
-(defoverload create-sink!  terra-workspace-sink-name create-terra-workspace-sink)
-(defoverload load-sink!    terra-workspace-sink-type load-terra-workspace-sink)
-(defoverload update-sink!  terra-workspace-sink-type update-terra-workspace-sink)
+(defmethod create-sink! terra-workspace-sink-name
+  [tx id request]
+  (write-terra-workspace-sink
+   tx id (terra-workspace-sink-validate-request-or-throw request)))
 
-(defoverload stage/done? terra-workspace-sink-type terra-workspace-sink-done?)
+(defoverload load-sink!   terra-workspace-sink-type load-terra-workspace-sink)
+(defoverload update-sink! terra-workspace-sink-type update-terra-workspace-sink)
+(defoverload stage/done?  terra-workspace-sink-type terra-workspace-sink-done?)
 
 (defoverload util/to-edn terra-workspace-sink-type terra-workspace-sink-to-edn)
+
+;; TerraDataRepo Sink
+(def ^:private ^:const datarepo-sink-name  "Terra DataRepo")
```

Comment:
Sorry, can you remind me why we're tagging these as `^:const` but not the other names, types, etc. for other stage implementations?

Reply:
They're constants so marking them as `^:const` allows the clojure compiler to inline them: https://stackoverflow.com/questions/9162558/how-does-clojure-const-work
I didn't change others because I wanted to limit scope drift.

Reply:
I've propagated `^:const`.